internal/parse/parse.go

```diff
@@ -43,16 +47,82 @@ var (
 	errDuplicateDest = errors.New("duplicate mount destination")
 )
 
+func mountIsReadWrite(m specs.Mount) bool {
+	// in case of conflicts, the last one wins, so it's not enough
+	// to check for the presence of either "rw" or "ro" anywhere
+	// with e.g. slices.Contains()
+	rw := true
+	for _, option := range m.Options {
+		switch option {
+		case "rw":
+			rw = true
+		case "ro":
+			rw = false
+		}
+	}
+	return rw
+}
+
+func convertToOverlay(m specs.Mount, store storage.Store, mountLabel, tmpDir string, uid, gid int) (specs.Mount, string, error) {
+	overlayDir, err := overlay.TempDir(tmpDir, uid, gid)
+	if err != nil {
+		return specs.Mount{}, "", fmt.Errorf("setting up overlay for %q: %w", m.Destination, err)
+	}
+	graphOptionsCopy := store.GraphOptions()
```

Comment:
`GraphOptions()` returns a slice, and assignment doesn't create a new underlying array, which is part of what makes slices faster than they might otherwise be.  You'd need something along the lines of what `copyStringSlice()` from our top-level `utils.go` does to do the equivalent of `slices.Clone()`.